src/util/file.rs

```diff
@@ -0,0 +1,128 @@
+use async_trait::async_trait;
+use rocket::fs::TempFile;
+use std::{ffi::OsStr, fs::DirEntry, path::Path};
+use tokio::io::AsyncReadExt;
+
+#[cfg(test)]
+use mockall::automock;
+
+#[cfg_attr(test, automock)]
```

Comment:
Automock, that is great!

Reply:
Yeah, I like it :+1: Unfortunately, it doesn't play particularly nice with traits that are both `async_trait` and have arguments with their own lifetime constraints (because both async_trait and automock mess with the lifetimes of the mocked trait), but that's work-around-able :v: